File: app/src/main/java/com/partycalc/database/Participant.java
package com.partycalc.database;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.os.Parcel;
import android.os.Parcelable;

@Entity
public class Participant implements Parcelable {

    @PrimaryKey(autoGenerate = true) int id;
    String name;
    int size;

    public static final Parcelable.Creator<Participant> CREATOR = new Parcelable.Creator<Participant>() {
        public Participant createFromParcel(Parcel in) {
            return new Participant(in);
        }

        @Override
        public Participant[] newArray(int size) {
            return new Participant[size];
        }
    };

    public Participant(String name, int size) {
        this.name = name;
        this.size = size;
    }

    public Participant(Parcel in) {
        this.id = in.readInt();
        this.name = in.readString();
        this.size = in.readInt();
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(getId());
        dest.writeString(getName());
        dest.writeInt(getSize());
    }
}

File: app/src/main/java/com/partycalc/database/Participant.kt
package com.partycalc.database

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class Participant : Parcelable {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    var name: String
    var size: Int

    constructor(name: String, size: Int) {
        this.name = name
        this.size = size
    }

    constructor(`in`: Parcel) {
        id = `in`.readInt()
        name = `in`.readString()
        size = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(name)
        dest.writeInt(size)
    }

    companion object {
        val CREATOR: Creator<Participant?> = object : Creator<Participant?> {
            override fun createFromParcel(`in`: Parcel): Participant? {
                return Participant(`in`)
            }

            override fun newArray(size: Int): Array<Participant?> {
                return arrayOfNulls(size)
            }
        }
    }
}